OemPkg/SecureBootRecovery/SecureBootRecovery.c

```diff
@@ -0,0 +1,136 @@
+/** @file
+  This application will attempt to append the 'Windows UEFI CA 2023' and then reboot the system.
+  On success, this application will allow the system to boot into 2023 signed Windows
+
+  Copyright (C) Microsoft Corporation
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Uefi.h>
+#include <Library/UefiApplicationEntryPoint.h>
+#include <Guid/ImageAuthentication.h>
+
+#include "RecoveryPayload.h"
+
+// 10 seconds in microseconds
+#define STALL_10_SECONDS  10000000
+
+/**
+  The user Entry Point for Application. The user code starts with this function
+  as the real entry point for the application.
+
+  @param[in] ImageHandle        The firmware allocated handle for the EFI image.
+  @param[in] SystemTable        A pointer to the EFI System Table.
+
+  @retval EFI_SUCCESS           The entry point is executed successfully.
+  @retval EFI_INVALID_PARAMETER SystemTable provided was not valid.
+  @retval other                 Some error occurs when executing this entry point.
+
+**/
+EFI_STATUS
+EFIAPI
+UefiMain (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  EFI_STATUS  Status;
+  UINT32      Attributes;
+
+  Attributes = EFI_VARIABLE_NON_VOLATILE |
+               EFI_VARIABLE_BOOTSERVICE_ACCESS |
+               EFI_VARIABLE_RUNTIME_ACCESS |
+               EFI_VARIABLE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS |
+               EFI_VARIABLE_APPEND_WRITE;

```

Comment:
This can be expressed more succinctly as:

```suggestion
  Attributes = VARIABLE_ATTRIBUTE_NV_BS_RT_AT | EFI_VARIABLE_APPEND_WRITE;
```

(with `#include <Guid/VariableFormat.h>`)